src/utils/fetchStream.ts

```diff
@@ -1,17 +1,61 @@
+import { Flag } from '$/features/flag/types'
+import { MiscAction } from '$/features/misc'
+import { RoomAction } from '$/features/room'
 import { RoomId } from '$/features/room/types'
+import { selectWalletProvider } from '$/features/wallet/selectors'
+import getRoomMetadata from '$/utils/getRoomMetadata'
+import { Provider } from '@web3-react/types'
+import { call, put, select } from 'redux-saga/effects'
 import type StreamrClient from 'streamr-client'
 import type { Stream } from 'streamr-client'
 
-export default async function fetchStream(roomId: RoomId, streamrClient: StreamrClient) {
-    try {
-        const stream: Stream = await streamrClient.getStream(roomId)
+export default function fetchStream(roomId: RoomId, streamrClient: StreamrClient) {
+    return call(function* () {
+        try {
+            const stream: Stream = yield streamrClient.getStream(roomId)
 
-        return stream
-    } catch (e: any) {
-        if (!/NOT_FOUND/.test(e?.message || '')) {
-            throw e
+            const {
+                tokenAddress,
+                tokenIds = [],
+                minRequiredBalance = '0',
+            } = getRoomMetadata(stream)
+
+            yield put(
```

Comment:
`fetchStreamr` which we use globally to fetch streams triggers gate caching. Every time it happens we know what's up. Possible overkill. Doesn't cost a lot.